sphereon-kmp-crypto-kms-azure/src/jvmMain/kotlin/com/sphereon/crypto/kms/azure/AzureKeyVaultCryptoProvider.jvm.kt

```diff
@@ -0,0 +1,389 @@
+        val keyType = alg?.toKeyType() ?: com.azure.security.keyvault.keys.models.KeyType.EC
+
+        val operations: Array<KeyOperation> = (keyOperations ?: arrayOf(KeyOperations.SIGN, KeyOperations.VERIFY))
+            .map {
+                it.toAzureKeyOperation()
+            }
+            .toTypedArray()
+
+        val createKeyOptions = CreateKeyOptions(keyName, keyType)
+            .setKeyOperations(*operations)
+
+        val keyVaultKey = withContext(Dispatchers.IO) {
+            keyClient.createKey(createKeyOptions).block()
+        }
+
+        if (keyVaultKey == null) {
+            logger.debug("Failed to create key in Azure Key Vault for reference: $keyName")
+            throw SignClientException("Failed to create key in Azure Key Vault")
+        }
+
+        val keyVaultJwk = keyVaultKey.toJwk()
+        val kid = keyVaultJwk.kid ?: generateJwkThumbprint(keyVaultJwk)
+        val jwk = keyVaultJwk.copy(kid = kid)
+        val publicCoseKey = CoseJoseKeyMappingService.toCoseKey(jwk)
+
+        return ManagedKeyPair(
+            kms = getId(),
+            kmsKeyRef = keyVaultKey.name,
+            kid = kid,
+            jose = JoseKeyPair(null, jwk),
+            cose = CoseKeyPair(null, publicCoseKey)
+        )
+    }
+
+    private fun determineSignatureAlgorithm(keyVaultKey: KeyVaultKey): com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm {
+        return when (keyVaultKey.key.keyType) {
+            com.azure.security.keyvault.keys.models.KeyType.RSA -> com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm.RS256
+            com.azure.security.keyvault.keys.models.KeyType.EC -> {
+                when (keyVaultKey.key.curveName) {
+                    KeyCurveName.P_256 -> com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm.ES256
+                    KeyCurveName.P_384 -> com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm.ES384
+                    KeyCurveName.P_521 -> com.azure.security.keyvault.keys.cryptography.models.SignatureAlgorithm.ES512
+                    else -> throw SignClientException("Unsupported curve: ${keyVaultKey.key.curveName}")
+                }
+            }
+            else -> throw SignClientException("Unsupported key type: ${keyVaultKey.key.keyType}")
+        }
+    }
+
+    override suspend fun createRawSignatureAsync(
+        keyInfo: IKeyInfo<*>,
+        input: ByteArray,
+        requireX5Chain: Boolean
+    ): ByteArray {
+        val keyVaultKey = withContext(Dispatchers.IO) {
+            keyClient.getKey(keyInfo.kmsKeyRef).block()
+        } ?: throw SignClientException("Key not found in Azure Key Vault for reference: ${keyInfo.kmsKeyRef}")
+
+        val cryptoClient = keyClient.getCryptographyAsyncClient(keyInfo.kmsKeyRef)
+
+        val signResult = withContext(Dispatchers.IO) {
+            cryptoClient.sign(determineSignatureAlgorithm(keyVaultKey), input.sha256()).block()
```

Comment:
The sha value (digest) should come from the keyInfo.signature alg by default. Only faillback to sha256